tests/scripts/task_kg_test.sh

```diff
@@ -36,28 +36,96 @@ export DGL_DOWNLOAD_DIR=${PWD}
 
 pushd $KG_DIR> /dev/null
 
-python3 -m nose -v --with-xunit tests/test_score.py || "run test_score.py on $1"
+python3 -m nose -v --with-xunit tests/test_score.py || fail "run test_score.py on $1"
 
 if [ "$2" == "cpu" ]; then
-    # verify CPU training
+    # verify CPU training DistMult
     python3 train.py --model DistMult --dataset FB15k --batch_size 128 \
         --neg_sample_size 16 --hidden_dim 100 --gamma 500.0 --lr 0.1 --max_step 100 \
-        --batch_size_eval 16 --valid --test -adv --eval_interval 30 --eval_percent 0.01
+        --batch_size_eval 16 --valid --test -adv --eval_interval 30 --eval_percent 0.01 \
+        --save_emb DistMult_FB15k_emb --data_path /data/kg || fail "run DistMult on $2"
+
+    # verify saving training result
+    python3 eval.py --model_name DistMult --dataset FB15k --hidden_dim 100 \
+        --gamma 500.0 --batch_size 16 --gpu -1 --model_path DistMult_FB15k_emb/ \
+        --eval_percent 0.01 --data_path /data/kg || fail "eval DistMult on $2"
+
+    # verify CPU training TransE
+    python3 train.py --model TransE --dataset FB15k --batch_size 128 \
+        --neg_sample_size 16 --hidden_dim 100 --gamma 24.0 --lr 0.1 --max_step 100 \
+        --batch_size_eval 16 --valid --test -adv --eval_interval 30 --eval_percent 0.01 \
+        --save_emb TransE_FB15k_emb --data_path /data/kg || fail "run TransE on $2"
+
+    # verify saving training result
+    python3 eval.py --model_name TransE --dataset FB15k --hidden_dim 100 \
+        --gamma 24.0 --batch_size 16 --gpu -1 --model_path TransE_FB15k_emb/ \
+        --eval_percent 0.01 --data_path /data/kg || fail "eval TransE on $2"
+
+    if [ "$1" == "pytorch" ]; then
+        # verify CPU training ComplEx
+        python3 train.py --model ComplEx --dataset FB15k --batch_size 128 \
+            --neg_sample_size 16 --hidden_dim 100 --gamma 500.0 --lr 0.1 --max_step 100 \
+            --batch_size_eval 16 --valid --test -adv --eval_interval 30 --eval_percent 0.01 \
+            --save_emb ComplEx_FB15k_emb --data_path /data/kg || fail "run ComplEx on $2"
+
+        # verify saving training result
+        python3 eval.py --model_name ComplEx --dataset FB15k --hidden_dim 100 \
+            --gamma 500.0 --batch_size 16 --gpu -1 --model_path ComplEx_FB15k_emb/ \
+            --eval_percent 0.01 --data_path /data/kg || fail "eval ComplEx on $2"
```

Comment:
I guess we don't need to verify all models in the CI. The correctness of the score functions is verified in the unit test. The reason to have these tests is to make sure the scripts can run. We don't verify the trained accuracy here. We do need a nightly test to verify the converged accuracy of the models.

Reply:
> I guess we don't need to verify all models in the CI. The correctness of the score functions is verified in the unit test. The reason to have these tests is to make sure the scripts can run. We don't verify the trained accuracy here. We do need a nightly test to verify the converged accuracy of the models.

Done